### PR TITLE
[AIRFLOW-6796] Clean up DAG serializations based on last_updated

### PR DESCRIPTION
**Pull request by MatthewRBruce**
DAG serializations were previous deleted based on whether the
DagFileProcessorManager had processed a particular python file.  This caused issues if DAGs were import via `globals()` via a different python file.  This
changes the deletion behaviour to be based on the last time a DAG was processed by the
scheduler instead.  

This also moves cleaning up of stale DAGs from `SchedulerJob` to `DagFileProcessorManager`  to support long running schedulers

---
Issue link: [AIRFLOW-6796](https://issues.apache.org/jira/browse/AIRFLOW-6796)

Make sure to mark the boxes below before creating PR: [x]

- [X] Description above provides context of the change
- [X] Commit message/PR title starts with `[AIRFLOW-NNNN]`. AIRFLOW-NNNN = JIRA ID<sup>*</sup>
- [X] Unit tests coverage for changes (not needed for documentation changes)
- [X] Commits follow "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)"
- [X] Relevant documentation is updated including usage instructions.
- [X] I will engage committers as explained in [Contribution Workflow Example](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#contribution-workflow-example).

<sup>*</sup> For document-only changes commit message can start with `[AIRFLOW-XXXX]`.

---
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in [UPDATING.md](https://github.com/apache/airflow/blob/master/UPDATING.md).
Read the [Pull Request Guidelines](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#pull-request-guidelines) for more information.

### CODE REVIEW

**Comment by kaxil** [comment]
Can you rebase on latest master please @MatthewRBruce

**Comment by MatthewRBruce** [comment]
Sure thing, I wasn't sure if this would still be needed with all the other serialization changes going on

**Comment by MatthewRBruce** [comment]
@kaxil I've rebased it now - I changed how the `DagCode` entries get cleaned up to be based on whether there are any DAGs in the DB with matching `fileloc`s instead of based on what `file_paths` were found.

**Review by kaxil — COMMENTED** [review]
So does this mean the DAG won't be deleted until it reaches `dag_cleanup_interval` even if the DAG file is removed? Or the DAG is updated, more tasks are added/deleted

**Comment by MatthewRBruce** [comment]
It won't delete the DAG until the last time the scheduler processed that DAG has surpassed `dag_cleanup_interval`. So yes, if the the DAG file was deleted, it will take around `dag_cleanup_interval`s for it to be removed.  I don't believe the DAG being updated or tasks being added/delete will come into play.

File: airflow/config_templates/config.yml [hunk 1]
```diff
@@ -1367,6 +1367,14 @@
       type: string
       example: ~
       default: "30"
+    - name: dag_cleanup_interval
+      description: |
+        How often unseen Dags should be marked inactive and their serializations deleted.
+        Setting to 0 will disable DAG cleanup.  Defaults to 5 minutes
```
Comment (kaxil) [inline airflow/config_templates/config.yml]:
```suggestion
        Setting to 0 will disable DAG cleanup. Defaults to 5 minutes
```

File: airflow/config_templates/default_airflow.cfg [hunk 2]
```diff
@@ -667,6 +667,10 @@ dag_dir_list_interval = 300
 # How often should stats be printed to the logs. Setting to 0 will disable printing stats
 print_stats_interval = 30
 
+# How often unseen Dags should be marked inactive and their serializations deleted.
+# Setting to 0 will disable DAG cleanup.  Defaults to 5 minutes
```
Comment (kaxil) [inline airflow/config_templates/default_airflow.cfg]:
```suggestion
# Setting to 0 will disable DAG cleanup. Defaults to 5 minutes
```

File: airflow/models/serialized_dag.py [hunk 3]
```diff
@@ -144,22 +144,22 @@ def remove_dag(cls, dag_id: str, session=None):
...
-        """Deletes DAGs not included in alive_dag_filelocs.
-
-        :param alive_dag_filelocs: file paths of alive DAGs
-        :param session: ORM Session
+    def remove_stale_dags(cls, expiration_date, session=None):
         """
-        alive_fileloc_hashes = [
-            DagCode.dag_fileloc_hash(fileloc) for fileloc in alive_dag_filelocs]
+        Deletes Serialized DAGs that were last touched by the scheduler before
+        the expiration date.  These DAGs were likely deleted.
```
Comment (kaxil) [inline airflow/models/serialized_dag.py]:
```suggestion
        the expiration date. These DAGs were likely deleted.
```

File: airflow/utils/dag_processing.py [hunk 4]
```diff
@@ -579,6 +579,12 @@ def __init__(self,
             )
             self._parallelism = 1
 
+        # How often to mark DAGs as inactive and delete their serializations
+        # if they haven't been processed recently.
+        self._dag_cleanup_interval = conf.getint('scheduler',
+                                                 'dag_cleanup_interval')
```
Comment (kaxil) [inline airflow/utils/dag_processing.py]:
```suggestion
        self._dag_cleanup_interval = conf.getint('scheduler', 'dag_cleanup_interval')
```

**Review by kaxil — APPROVED** [review]

**Pull request merged**